src/openloco/config.convert.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "config.h"
+
+#ifdef _WIN32
+// Ignore warnings generated from yaml-cpp
+#pragma warning(push)
+#pragma warning(disable : 4127) // conditional expression is constant
+#pragma warning(disable : 4251) // 'identifier': 'object_type1' 'identifier1' needs to have dll-interface to be used by clients of 'object_type' 'identfier2'
+#pragma warning(disable : 4275) // non dll-interface 'classkey' 'identifier1' used as base for dll-interface 'classkey' 'identifier2'
+#pragma warning(disable : 4996) // declaration deprecated
```

Comment:
Do these have to be used each time yaml is used? What are the issue/PR numbers for fixing them upstream? Till they get merged, can we have a proxy header that does all those warnings for us?

Reply:
* Some may be shared with gcc (I don't know): https://github.com/jbeder/yaml-cpp/issues/601
* The deprecated one only happens when targeting C++17.
* The others, not sure what they really mean but they only happen with W4 or above I think.

Reply:
Maybe it would be nice to create one for filesystem too, good point. The problem With filesystem is we need the correct types and namespaces, which we already partially solve with
```cpp
#ifdef _OPENLOCO_USE_BOOST_FS_
namespace fs = boost::filesystem;
#else
namespace fs = std::experimental::filesystem;
#endif
```
but due to API differences we have to use stuff like
```cpp
#ifdef _OPENLOCO_USE_BOOST_FS_
            cfg.loco_install_path = path.make_preferred().string();
#else
            cfg.loco_install_path = path.make_preferred().u8string();
#endif
```
and
```cpp
#ifdef _OPENLOCO_USE_BOOST_FS_
    boost::system::error_code ec;
#else
    std::error_code ec;
#endif
```

Our header should address at least the latter part and we could look into solving the former problem too.